packages/modules/viewer-react/src/hooks/useUnifiedSelectionSync.tsx

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import React from "react";
+import { SchemaContext } from "@itwin/ecschema-metadata";
+import { SelectionStorage, enableUnifiedSelectionSyncWithIModel } from "@itwin/unified-selection";
+import { createCachingECClassHierarchyInspector } from "@itwin/presentation-shared";
+import { createECSchemaProvider, createECSqlQueryExecutor } from "@itwin/presentation-core-interop";
+import { IModelConnection } from "@itwin/core-frontend";
+import { Presentation } from "@itwin/presentation-frontend";
+
+type SelectionScope = ReturnType<Parameters<typeof enableUnifiedSelectionSyncWithIModel>[0]["activeScopeProvider"]>;
+
+interface UseUnifiedSelectionSyncProps {
+  iModelConnection?: IModelConnection;
+  selectionStorage?: SelectionStorage;
+  getSchemaContext?: (imodel: IModelConnection) => SchemaContext;
+}
+
+export function useUnifiedSelectionSync({ iModelConnection, selectionStorage, getSchemaContext }: UseUnifiedSelectionSyncProps) {
+  React.useEffect(() => {
+    if (!iModelConnection || !selectionStorage || !getSchemaContext) {
+      return;
+    }
+    const schemaContext = getSchemaContext(iModelConnection);
+    return enableUnifiedSelectionSyncWithIModel({
+      imodelAccess: {
+        ...createECSqlQueryExecutor(iModelConnection),
+        ...createCachingECClassHierarchyInspector({ schemaProvider: createECSchemaProvider(schemaContext) }),
+        key: iModelConnection.key,
+        hiliteSet: iModelConnection.hilited,
+        selectionSet: iModelConnection.selectionSet,
+      },
+      selectionStorage,
+      activeScopeProvider: getActiveScope,
+    });
+  }, [iModelConnection, selectionStorage]);
```

Comment:
is the dep array here missing `getSchemaContext`?

Reply:
yes, added